test/functional/sample_types_controller_test.rb

```diff
@@ -138,6 +138,47 @@ class SampleTypesControllerTest < ActionController::TestCase
     assert_equal sample_type, sample_type.sample_attributes.last.linked_sample_type
   end
 
+  test 'create with creators' do
+    creator = Factory(:person)
+    assert_difference('SampleType.count') do
+      post :create, params: { sample_type: { title: 'Hello!',
+                                             project_ids: @project_ids,
+                                             description: 'The description!!',
+                                             contributor: @person,
+                                             creator_ids: [creator.id],
+                                             other_creators: 'John Smith, Jane Smith',
+                                             sample_attributes_attributes: {
+                                               '0' => {
+                                                 pos: '1', title: 'a string', required: '1', is_title: '1',
+                                                 sample_attribute_type_id: @string_type.id, _destroy: '0'
+                                               }
+                                             } } }
+    end
+    type = assigns(:sample_type)
+    assert_equal @person, type.contributor
+    assert_equal creator.id, type.creator_ids.first
```

Comment:
this would be better as

`assert_equal [creator], type.creators`

just incase there are more than 1 creators in the list